Canary 9.4.5-canary.36 => 9.4.5-canary.37 - can't resolve main.js

# Bug report

Sorry @timneutkens @Timer don't have time this morning for full-repro - but figured you'd want to know. 

Getting this just on simple build run between 36 => 37 canary

Wild guess - but could it be related to here? https://github.com/vercel/next.js/pull/15224/files#diff-4ec59a55144129a45f46da71c0777df9

```js
 ModuleNotFoundError: Module not found: Error: Can't resolve 'main.js' in '/vercel/419420aa'
```

## Describe the bug

Application won't build between the two versions

@timneutkens my hunch was correct - here you go https://github.com/followbl/pwa-bug

Steps to reproduce

1. run `yarn run build`

result

`ModuleNotFoundError: Module not found: Error: Can't resolve 'main.js' in`

expected result - no error on build

I'll report this in next-pwa incase there are any thoughts or helpful tips there. thank you as always!

--- a/packages/next/build/webpack-config.ts
+++ b/packages/next/build/webpack-config.ts
@@ -1278,7 +1278,10 @@ export default async function getBaseWebpackConfig(
         const originalFile = clientEntries[
           CLIENT_STATIC_FILES_RUNTIME_MAIN
         ] as string
-        entry[CLIENT_STATIC_FILES_RUNTIME_MAIN] = [...['main.js'], originalFile]
+        entry[CLIENT_STATIC_FILES_RUNTIME_MAIN] = [
+          ...entry['main.js'],
+          originalFile,
+        ]
       }
       delete entry['main.js']
 

--- a/test/integration/webpack-config-mainjs/client/polyfills.js
+++ b/test/integration/webpack-config-mainjs/client/polyfills.js
@@ -1,0 +1,8 @@
+/* eslint no-extend-native: 0 */
+
+// This file runs before React and Next.js core
+// This file is loaded for all browsers
+// Next.js includes a number of polyfills only for older browsers like IE11
+// Make sure you don't duplicate these in this file
+// https://github.com/vercel/next.js/blob/canary/packages/next-polyfill-nomodule/src/index.js
+console.log('Load your polyfills')

--- a/test/integration/webpack-config-mainjs/next.config.js
+++ b/test/integration/webpack-config-mainjs/next.config.js
@@ -1,0 +1,19 @@
+module.exports = {
+  webpack: function (cfg) {
+    const originalEntry = cfg.entry
+    cfg.entry = async () => {
+      const entries = await originalEntry()
+
+      if (
+        entries['main.js'] &&
+        !entries['main.js'].includes('./client/polyfills.js')
+      ) {
+        entries['main.js'].unshift('./client/polyfills.js')
+      }
+
+      return entries
+    }
+
+    return cfg
+  },
+}

--- a/test/integration/webpack-config-mainjs/pages/static.js
+++ b/test/integration/webpack-config-mainjs/pages/static.js
@@ -1,0 +1,3 @@
+export default function StaticPage() {
+  return <>Hello World</>
+}

--- a/test/integration/webpack-config-mainjs/test/index.test.js
+++ b/test/integration/webpack-config-mainjs/test/index.test.js
@@ -1,0 +1,15 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import { nextBuild } from 'next-test-utils'
+
+jest.setTimeout(1000 * 60 * 2)
+const appDir = join(__dirname, '../')
+
+describe('Customized webpack config with main.js', () => {
+  it('should run correctly with main.js customized', async () => {
+    const { code } = await nextBuild(appDir, [], {})
+
+    expect(code).toBe(0)
+  })
+})

